### ISSUE TITLE
(Probably) frequent name clashes

### ISSUE THREAD

**Issue by m-toman** [MycroftAI/mimic#83]
This is not really an issue with mimic but probably still worth considering:
I found that the function names in src/audio/audio.c frequently clash with other libraries (e.g. Acapela TTS). Perhaps prefixing is an option here?

Also, it seems that when defining CST_AUDIO_NONE (and e.g. using another audio framework), only audio_stream_chunk in https://github.com/MycroftAI/mimic/blob/master/src/audio/au_streaming.c requires audio.c - and it seems to be an example function only.
So probably surrounding audio_stream_chunk with #ifndef CST_AUDIO_NONE would make sense? (or even move that function to an example file).

**Comment by forslund** [comment]
Thanks for reporting.
Prefixing is a good solution, I guess the `mimic_` prefix is what should be used since it's part of the library interface? Do you feel like making a PR with this change?

Regarding the `au_streaming.c` file it is used by the mimic command line application if the last argument is `stream`. I think it's organized in this way to not clutter the main application with audio-if-defs.

I've only briefly looked at the streaming bits and haven't got the faintest idea what it's for so I could be totally wrong.

**Comment by m-toman** [comment]
OK, I'll prefix them and the calls with `mimic_`.

I've used the streaming synthesis and needed `au_streaming.c` for `cst_audio_streaming_info`.
I did not use the `audio_stream_chunk` function because I only need the synthesized waveforms but no audio playing capabilities (so I need audio.c only because of this function).
But I don't have any other suggestions either on that (but my personal problem will be solved by the prefixing ;))
